var/spack/repos/builtin/packages/swak4foam/package.py

```diff
@@ -0,0 +1,210 @@
+##############################################################################
+# Copyright (c) 2017 Mark Olesen, OpenCFD Ltd.
+#
+# This file was authored by Mark Olesen <mark.olesen@esi-group.com>
+# and is released as part of spack under the LGPL license.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for the LLNL notice and the LGPL.
+#
+# License
+# -------
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+#
+# Legal Notice
+# ------------
+# OPENFOAM is a trademark owned by OpenCFD Ltd
+# (producer and distributor of the OpenFOAM software via www.openfoam.com).
+# The trademark information must remain visible and unadulterated in this
+# file and via the "spack info" and comply with the term set by
+# http://openfoam.com/legal/trademark-policy.php
+#
+# This file is not part of OpenFOAM, nor does it constitute a component of an
+# OpenFOAM distribution.
+#
+##############################################################################
+from spack import *
+from spack.environment import *
+import llnl.util.tty as tty
+
+import multiprocessing
+import os
+from os.path import isfile
+
+
+#: Could move this out to openfoam-com utilities
+def find_foam_projectdir(foam_spec):
+    """Find the directory containing the OpenFOAM etc/bashrc file.
+    This directory corresponds to the WM_PROJECT_DIR.
+    Use a maxdepth of 1 in the search (for the directory).
+    Return None on failure.
+    """
+    testfile = join_path("etc", "bashrc")
+
+    if isfile(join_path(foam_spec.prefix, testfile)):
+        return foam_spec.prefix   # Trivial: located in prefix directly
+
+    # Check first level of sub-directories
+    with working_dir(foam_spec.prefix):
+        for d in os.listdir("."):
+            if not d.startswith('.') and isfile(join_path(d, testfile)):
+                return join_path(foam_spec.prefix, d)
+
+    raise RuntimeError('No OpenFOAM project directory found')
+    return None
```

Comment:
It might be easier to set this in the openfoam providers. This would look like:
```python
def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
    spack_env.set('WM_PROJECT_DIR', join_path(self.prefix, 'etc', 'bashrc'))
```
You can do this for as many variables as you want.